### PR TITLE
Apim 1520 entrypoint add with context path

### PR DESCRIPTION
**Pull request by loriepisicchio**
## Issue

https://gravitee.atlassian.net/browse/APIM-1520

## Description

Handle adding an HTTP entrypoint to an API that only has SUBSCRIPTION listener

Also include some small UI improvements + bugfix

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jyljvbeesz.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/entrypoints-v4/api-entrypoints-v4-general.component.html [hunk 1]
```diff
@@ -89,5 +82,16 @@ <h3>Entrypoints</h3>
         Add an entrypoint
       </button>
     </div>
+
+    <div class="entrypoints__footer">
+      <button
+        mat-flat-button
+        color="primary"
+        type="submit"
+        [disabled]="((formGroup.pristine || formGroup.invalid) && entrypointToBeRemoved.length === 0) || dataSource.length === 0"
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/entrypoints-v4/api-entrypoints-v4-general.component.html]: What do you think to trying to use only the `formGroup.pristine || formGroup.invalid` ? 
If we delete something, the form should know about it.

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/entrypoints-v4/api-entrypoints-v4-general.component.html]: ha ok it's not a form

Comment (loriepisicchio) [reply gravitee-apim-console-webui/src/management/api/entrypoints-v4/api-entrypoints-v4-general.component.html]: No it's not but it's a good idea to add another FormControl for this. That will simplify this condition.

File: gravitee-apim-console-webui/src/management/api/entrypoints-v4/edit/api-entrypoints-v4-add-dialog.component.html [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
+</ng-container>
+<ng-template #contextPathForm>
+  <div class="header">
+    <div class="header__title">Configure your context-paths</div>
+    <p class="header__subtitle">Configure at least one context-path for your API before adding this entrypoint</p>
   </div>
-</form>
+  <form *ngIf="contextPathFormGroup" [formGroup]="contextPathFormGroup" (ngSubmit)="saveWithContextPath()">
+    <gio-form-listeners-context-path formControlName="contextPath"></gio-form-listeners-context-path>
+    <div class="footer">
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/entrypoints-v4/edit/api-entrypoints-v4-add-dialog.component.html]: I wonder why you don't use 
`<mat-dialog-actions class="actions" align="end">` or `</mat-dialog-content>`

I haven't seen if the buttons are aligned to the right, but that would be preferable for the uniqueness of the screens.

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**